latch_cli/services/local_dev.py

```diff
@@ -353,68 +312,41 @@ def file_filter(filename: str) -> bool:
                         "Successfully connected to remote instance."
                     )
                     try:
-                        while True:
-                            with session.input.raw_mode():
-                                cmd: str = await session.prompt_async()
-                                cmd = cmd.strip()
-
-                            if cmd == "":
-                                continue
-
-                            if cmd in QUIT_COMMANDS:
-                                raise EOFError
-
-                            if not COMMAND_EXPRESSION.match(cmd):
-                                await aioconsole.aprint("Invalid command")
-
-                                closest = difflib.get_close_matches(
-                                    cmd.strip().split()[0], COMMANDS, 1
-                                )
-                                if len(closest) > 0:
-                                    await aioconsole.aprint(
-                                        f"Did you mean \x1b[1m{closest[0]}\x1b[22m?"
-                                    )
-                                continue
-
-                            if (
-                                cmd.startswith("run")
-                                or cmd == "list-tasks"
-                                or cmd == "shell"
-                            ):
-                                await aioconsole.aprint("Syncing local changes... ")
-                                await _copy_files(ssh_conn, pkg_root)
-                                await aioconsole.aprint("Finished syncing.")
-
-                            await _send_message(ws, cmd)
-
-                            image_name_tagged = _get_latest_image(pkg_root)
-                            await _send_message(ws, docker_access_token)
-                            await _send_message(ws, image_name_tagged)
-                            await aioconsole.aprint(f"Pulling {image_name_tagged}... ")
-
-                            await _get_messages(ws, show_output=False)
-                            await _get_messages(ws, show_output=False)
-                            await aioconsole.aprint("Image successfully pulled.\n")
-
-                            await _get_messages(ws, show_output=True)
-
-                            if cmd == "shell":
-                                with session.input.detach():
-                                    with patch_stdout(raw=True):
-                                        await _shell_session(ws)
-                                        await ws.drain()
+                        await aioconsole.aprint("Setting up local sync...")
+                        asyncio.create_task(_poll_rsync(pkg_root, centromere_ip, ssh_port, key_path))
```

Comment:
await the task at some point, otherwise you will lose all exceptions etc

Reply:
It is an infinite loop task so I cannot await it.